src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,166 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationException;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static final RegistrationServiceImpl REGISTRATION_SERVICE
+            = new RegistrationServiceImpl();
+    private static final StorageDaoImpl STORAGE_DAO = new StorageDaoImpl();
+    private StorageDaoImpl storageDao = new StorageDaoImpl();
+
+    @BeforeEach
+    void setUp() {
+        storageDao = new StorageDaoImpl();
+        Storage.people.clear();
```

Comment:
Good use of `@BeforeEach` to clear the storage before each test. This ensures that tests are independent of each other.